wrappers/awskms/awskms.go

```diff
@@ -277,48 +283,49 @@ func (k *Wrapper) Decrypt(_ context.Context, in *wrapping.BlobInfo, opt ...wrapp
 }
 
 // Client returns the AWS KMS client used by the wrapper.
-func (k *Wrapper) Client() kmsiface.KMSAPI {
+func (k *Wrapper) Client() KmsApi {
 	return k.client
 }
 
 // GetAwsKmsClient returns an instance of the KMS client.
-func (k *Wrapper) GetAwsKmsClient() (*kms.KMS, error) {
-	credsConfig := &awsutil.CredentialsConfig{}
-
-	credsConfig.AccessKey = k.accessKey
-	credsConfig.SecretKey = k.secretKey
-	credsConfig.SessionToken = k.sessionToken
-	credsConfig.Filename = k.sharedCredsFilename
-	credsConfig.Profile = k.sharedCredsProfile
-	credsConfig.RoleARN = k.roleArn
-	credsConfig.RoleSessionName = k.roleSessionName
-	credsConfig.WebIdentityTokenFile = k.webIdentityTokenFile
-	credsConfig.Region = k.region
-	credsConfig.Logger = k.logger
-
-	credsConfig.HTTPClient = cleanhttp.DefaultClient()
-
-	creds, err := credsConfig.GenerateCredentialChain()
+func (k *Wrapper) GetAwsKmsClient(ctx context.Context) (*kms.Client, error) {
+	credsConfig := &awsutil.CredentialsConfig{
+		AccessKey:            k.accessKey,
+		SecretKey:            k.secretKey,
+		SessionToken:         k.sessionToken,
+		Filename:             k.sharedCredsFilename,
+		Profile:              k.sharedCredsProfile,
+		RoleARN:              k.roleArn,
+		RoleSessionName:      k.roleSessionName,
+		WebIdentityTokenFile: k.webIdentityTokenFile,
+		Region:               k.region,
+		Logger:               k.logger,
+		HTTPClient:           cleanhttp.DefaultClient(),
+	}
+
+	creds, err := credsConfig.GenerateCredentialChain(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	awsConfig := &aws.Config{
-		Credentials: creds,
-		Region:      aws.String(credsConfig.Region),
-		HTTPClient:  cleanhttp.DefaultClient(),
+	clientOpts := []func(*config.LoadOptions) error{
+		config.WithCredentialsProvider(creds.Credentials),
+		config.WithRegion(k.region),
+		config.WithHTTPClient(cleanhttp.DefaultClient()),
 	}
 
 	if k.endpoint != "" {
-		awsConfig.Endpoint = aws.String(k.endpoint)
+		clientOpts = append(clientOpts, config.WithBaseEndpoint(k.endpoint))
 	}
 
-	sess, err := session.NewSession(awsConfig)
+	cfg, err := config.LoadDefaultConfig(ctx, clientOpts...)
 	if err != nil {
 		return nil, err
 	}
 
-	client := kms.New(sess)
+	kmsOpts := []func(*kms.Options){}
+
+	client := kms.NewFromConfig(cfg, kmsOpts...)
```

Comment:
Do we intend to add any options here? I think we can just omit this variable if not.